edge-modules/edge-dashboard/src/status.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use std::net::TcpStream;
+use std::process::Command;
+
+use actix_web::error::ErrorInternalServerError;
+use actix_web::Error as ActixError;
+use actix_web::*;
+
+use crate::state::{return_response, Device};
+
+pub fn get_state(device: web::Data<Option<Device>>) -> HttpResponse {
+    if let Some(dev) = device.get_ref() {
```

Comment:
I noticed that you have multiple places where you are using `if let ... else ... ` constructs. It's not necessarily a bad thing it works fine obviously. But it looks like c code, not idiomatical rust code. Rust employs pattern matching for it.